tests/fsharp/typecheck/sigs/neg24.fs

```diff
@@ -69,3 +69,245 @@ type C() =
+        [ for i in 1 .. 10 do 
+             10
+             printfn "hello" ] 
+
+    // Another check that implicit yield is enabled
+    let v5 =
+        [ if true then 1 ] 
+
+    // Another check that implicit yield is enabled
+    let v5a = 
+        [ printfn "hello";
+          if true then 1 ] 
+
+    // Another check that implicit yield is enabled
+    let v5b =
+        [ if true then 
+            printfn "hello"
+            1 
+        ] 
+
+module ArrayPositive2 = 
+    let a3 = 
+        [| (if true then 1 else 2) |] // simple single element sequence 
+
+    let a5 = 
+        [| if true then 1 |] 
+
+    let l10 = 
+        [ printfn "hello"; yield 1; yield 2 ] // expect ok - the printfn has type unit and is not interpreted as a yield
+    
+    // simple case of explicit yield
+    let l12 : int list = 
+        [ printfn "hello"
+          if true then yield 1 else yield 2 ] 
+
+    // check subsumption is alloed when using explicit yield
+    let l13 : obj list = 
+        [ printfn "hello"
+          if true then yield 1 else yield 2 ] 
+
+    // check subsumption is alloed when using explicit yield
+    let l14 : obj list = 
+        [ printfn "hello"
+          if true then yield 1 ] 
+
+module SeqPositive2 = 
+    let s2 = 
+        seq { if true then () else () } 
+
+    let s6 = 
+        seq { if true then () } 
+    
+    let s4 = 
+        seq { if true then 1 else 2 }
+
+    let s6 = 
+        seq { if true then 1  }
+    
+    let s7 = 
+        seq { match 1 with 1 -> 4 | 2 -> 5 | 3 -> 6 | _ -> ()  }
+    
+module BuilderPositive2 = 
+    type L<'T> = { Make: (unit -> 'T list) }
+    let L f = { Make = f }
+
+    type ListBuilder() = 
+        member __.Combine(x1: L<'T>, x2: L<'T>) = L(fun () -> x1.Make() @ x2.Make())
+        member __.Delay(f: unit -> L<'T>) = L(fun () -> f().Make())
+        member __.Zero() = L(fun () -> [])
+        member __.Yield(a: 'T) = L(fun () -> [a])
+        member __.YieldFrom(x: L<'T>) = x
+
+    let list = ListBuilder()
+    let empty<'T> : L<'T> = list.Zero()
+
+    let v3 =
+        list { if true then 1 else 2 }  // implicit yield enabled
+
+    let v3y =
+        list { if true then yield 1 else yield 2 }  // equivalent explicit yield
+
+    let v3a = 
+        list {
+          printfn "hello"
+          if true then 1 else 2  // implicit yield enabled
+        }
+
+    let v3ay = 
+        list {
+          printfn "hello"
+          if true then yield 1 else yield 2  // equivalent explicit yield 
+        }
+
+    let v3b = 
+        list {
+          printfn "hello"
+          if true then 1 else 2 // implicit yield, even though there is a `yield!`
+          yield! empty
+        }
+
+    let v3bc = 
+        list {
+          printfn "hello"
+          if true then yield 1 else yield 2 // equivalent explicit yield
+          yield! empty
+        }
+
+    // Note, no subsumption is permitted when using yield in computation expressions, with or eithout implicit yield
```

Comment:
Typo: `eithout` --> `without`

Reply:
Not sure how I feel about demonstrating this via a comment here. It almost implies that we're ignoring a test case

Reply:
I added this as a negative test case, thanks